File: c++/operators/operators_bitwise_shift.cpp
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */
// SPDX-License-Identifier: GPL-3.0-or-later

#include <iostream>

int main()
{
  int n = 4; // 00100

  std::cout << n << " >> 1 = " << (n >> 1) << std::endl; // 00010 // 2
  std::cout << n << " >> 2 = " << (n >> 2) << std::endl; // 00001 // 1
  std::cout << n << " << 1 = " << (n << 1) << std::endl; // 01000 // 8
  std::cout << n << " << 2 = " << (n << 2) << std::endl; // 10000 // 16
}

File: python/operators/operators_bitwise_shift.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

n = 4 # 00100

print("{} >> 1 = {}".format(n, (n >> 1))) # 00010 # 2
print("{} >> 2 = {}".format(n, (n >> 2))) # 00001 # 1
print("{} << 1 = {}".format(n, (n << 1))) # 01000 # 8
print("{} << 2 = {}".format(n, (n << 2))) # 10000 # 16
